content/blog/2019/11/2019-11-21-welcome-to-the-matrix.adoc

```diff
@@ -265,15 +265,15 @@ matrix {
 }
 ----
 
-
 Occasionally I run my pipeline manually from the Jenkins Web UI. 
-When I do that, I'd like to be able to select just one platform to run, but I can't use `excludes` for this.
+When I do that, I'd like to be able to select just one platform to run.
 The `axis` and `exclude` directives define the static set of cells that make up the matrix. 
-That set of combinations is generated before the start of the run, before any parameters. 
+That set of combinations is generated before the start of the run, before any parameters are processed. 
+What this means is that I can't add or remove cells from a matrix after the job has started.
 
-So, I can't add or remove cells from a matrix after the job has started but I can use a "per-cell" directives. 
-I'll add a `choice` parameter with the list of platforms and use a `when` directive inside `matrix` to control which cells in the matrix are run.
-The log below shows a portion of the output when I select `PLATFORM_FILTER = 'mac'`.
+The "per-cell" directives, on the other hand, are evaluated at runtime. 
+I can use the "per-cell" `when` directive inside `matrix` to control which cells in the matrix are executed.
+I'll add a `choice` parameter with the list of platforms and add conditions to `when` directive that will either let all platforms execute or only execute cells that match my selected platform. 
```

Comment:
Really good.

Last sentence: when I read it out loud, it feels like it needs a couple of commas in it. Also added "the" before `when`.

I'll add a `choice` parameter with the list of platforms, and add conditions to the `when` directive, which will either let all platforms execute, or only execute cells that match my selected platform.